examples/bri-3/src/bri/identity/bpiSubjects/persistence/bpiSubjects.repository.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common'
+import { PrismaService } from 'prisma/prisma.service';
+import { BpiSubject } from '../models/bpiSubject';
+
+// Repositories are the only places that talk the Prisma language of models.
+// They are always mapped to and from domain objects so that the business layer of the application
+// does not have to care about the ORM.
+@Injectable()
+export class BpiSubjectRepository extends PrismaService {
+    async getBpiSubjectById(id: string): Promise<BpiSubject> {
+        const bpiSubjectModel = await this.bpiSubject.findUnique({ where: { id: id } });
+        return new BpiSubject( // TODO: Write generic mapper prismaModel -> domainObject
```

Comment:
i could be wrong with this, but will a lot of mappers introduce a lot of boilerplate code
prisma model -> domain object -> dto object, and in most cases they are just same objects

Reply:
That is why is must be a generic mapper. Usually something along the line of: Object.assign(this). This way it is just a single line of code that does the mapping. Domain objects will not be the same, they ll contain methods for the execution of  the business logic. Persistence model might contain fields such as CreatedAt,  ModifiedAt that will not be mapped to the domain object as they are there only for audit purposes.